src/main/java/org/opensearch/agent/tools/CreateAnomalyDetectorTool.java

```diff
@@ -0,0 +1,452 @@
+                if (dataAsMap == null) {
+                    listener.onFailure(new IllegalStateException("Remote endpoint fails to inference."));
+                    return;
+                }
+                String finalResponse = dataAsMap.get("response");
+                if (Strings.isNullOrEmpty(finalResponse)) {
+                    listener.onFailure(new IllegalStateException("Remote endpoint fails to inference, no response found."));
+                    return;
+                }
+
+                // use regex pattern to extract the suggested parameters for the create anomaly detector API
+                Pattern pattern = Pattern.compile(EXTRACT_INFORMATION_REGEX);
+                Matcher matcher = pattern.matcher(finalResponse);
+                if (!matcher.matches()) {
+                    log
+                        .error(
+                            "The inference result from remote endpoint is not valid because the result: ["
+                                + finalResponse
+                                + "] cannot match the regex: "
+                                + EXTRACT_INFORMATION_REGEX
+                        );
+                    listener
+                        .onFailure(
+                            new IllegalStateException(
+                                "The inference result from remote endpoint is not valid, cannot extract the key information from the result."
+                            )
+                        );
+                    return;
+                }
+
+                // remove double quotes or whitespace if exists
+                String categoryField = matcher.group(1).replaceAll("\"", "").strip();
+                String aggregationField = matcher.group(2).replaceAll("\"", "").strip();
+                String aggregationMethod = matcher.group(3).replaceAll("\"", "").strip();
+
+                Map<String, String> result = ImmutableMap
+                    .of(
+                        OUTPUT_KEY_INDEX,
+                        indexName,
+                        OUTPUT_KEY_CATEGORY_FIELD,
+                        categoryField,
+                        OUTPUT_KEY_AGGREGATION_FIELD,
+                        aggregationField,
+                        OUTPUT_KEY_AGGREGATION_METHOD,
+                        aggregationMethod,
+                        OUTPUT_KEY_DATE_FIELDS,
+                        dateFieldsJoiner.toString()
+                    );
+                listener.onResponse((T) AccessController.doPrivileged((PrivilegedExceptionAction<String>) () -> gson.toJson(result)));
+            }, e -> {
+                log.error("fail to predict model: " + e);
+                listener.onFailure(e);
+            }));
+        }, e -> {
+            log.error("fail to get mapping: " + e);
+            String errorMessage = e.getMessage();
+            if (errorMessage.contains("no such index")) {
```

Comment:
It's better to check exception type instead of error message, since error message isn't a contract and might change in future.

Reply:
Changed that, please help to take a look, thank you!